spec/app/domain/authentication/authn-azure/validate_application_identity_spec.rb

```diff
@@ -0,0 +1,433 @@
+  let(:xms_mirid_token_missing_resource_groups_field) { "/subscriptions/some-subscription-id-value/providers/Microsoft.ManagedIdentity/some-user-assigned-identity-value" }
+  let(:xms_mirid_token_missing_providers_field) { "/subscriptions/some-subscription-id-value/resourcegroups/some-resource-group-value/" }
+  let(:xms_mirid_token_missing_initial_slash) { "subscriptions/some-subscription-id-value/resourcegroups/some-resource-group-value/providers/Microsoft.ManagedIdentity/userAssignedIdentities/some-system-assigned-identity-value" }
+
+  let(:invalid_xms_mirid_token_providers_field) { "/subscriptions/some-subscription-id-value/resourcegroups/some-resource-group-value/providers/Microsoft.ManagedIdentity/some-user-assigned-identity-value" }
+
+  let(:oid_token_field) { "test-oid" }
+
+  let(:system_assigned_identity_annotation) { double("SystemAssignedIdentityAnnotation") }
+
+  let(:mismatched_subscription_id_annotation) { double("MismatchedSubscriptionIdAnnotation") }
+  let(:mismatched_resource_group_annotation) { double("MismatchedResourceGroupAnnotation") }
+  let(:mismatched_system_assigned_identity_annotation) { double("MismatchedSystemAssignedAnnotation") }
+  let(:mismatched_user_assigned_identity_annotation) { double("MismatchedUserAssignedAnnotation") }
+
+  let(:validate_azure_annotations) { double("ValidateAzureAnnotations") }
+
+  before(:each) do
+    allow(host).to receive(:annotations)
+                     .and_return([subscription_id_annotation, resource_group_annotation])
+
+    allow(validate_azure_annotations).to receive(:call)
+                                           .and_return(true)
+
+    allow(resource_class).to receive(:[])
+                               .with("#{account}:host:#{hostname}")
+                               .and_return(host)
+
+    define_host_annotation(system_assigned_identity_annotation, "#{global_annotation_type}/system-assigned-identity", oid_token_field)
+    define_host_annotation(mismatched_subscription_id_annotation, "#{global_annotation_type}/subscription-id", "mismatched-subscription-id")
+    define_host_annotation(mismatched_resource_group_annotation, "#{global_annotation_type}/resource-group", "mismatched-resource-group")
+    define_host_annotation(mismatched_user_assigned_identity_annotation, "#{global_annotation_type}/user-assigned-identity", "mismatched-user-assigned-identity")
+    define_host_annotation(mismatched_system_assigned_identity_annotation, "#{global_annotation_type}/system-assigned-identity", "mismatched-system-assigned-identity")
+  end
+
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
+
+  context "A valid xms_mirid claim in the Azure token" do
+
+    context "that does not begin with a /" do
```

Comment:
I would move this below. Although it is a vanilla flow it is not the straight-forward one. Also, why isn't this inside `"with a valid application identity"`? Doesn't it have a valid app id?
I would have:
```
context "A valid xms_mirid claim in the Azure token" do
    context "with a valid application identity" do
      context "and an Azure token with matching data" do
        context "with no assigned Azure identity in application identity" do
          subject do
            Authentication::AuthnAzure::ValidateApplicationIdentity.new(
              resource_class:             resource_class,
              validate_azure_annotations: validate_azure_annotations,
              ).call(
              account:               account,
              service_id:            service_id,
              username:              username,
              xms_mirid_token_field: xms_mirid_token_field_system_assigned_identity,
              oid_token_field:       oid_token_field
            )
          end

          it "does not raise an error" do
            expect { subject }.to_not raise_error
          end
        end

        context "with a user assigned Azure identity in application identity" do
          before(:each) do
            allow(host).to receive(:annotations)
                             .and_return([subscription_id_annotation, resource_group_annotation, user_assigned_identity_annotation])
          end
          subject do
            Authentication::AuthnAzure::ValidateApplicationIdentity.new(
              resource_class:             resource_class,
              validate_azure_annotations: validate_azure_annotations,
              ).call(
              account:               account,
              service_id:            service_id,
              username:              username,
              xms_mirid_token_field: xms_mirid_token_field_user_assigned_identity,
              oid_token_field:       oid_token_field
            )
          end

          it "does not raise an error" do
            expect { subject }.to_not raise_error
          end
        end

        context "with a system assigned Azure identity in application identity" do
          subject do
            Authentication::AuthnAzure::ValidateApplicationIdentity.new(
              resource_class:             resource_class,
              validate_azure_annotations: validate_azure_annotations,
              ).call(
              account:               account,
              service_id:            service_id,
              username:              username,
              xms_mirid_token_field: xms_mirid_token_field_system_assigned_identity,
              oid_token_field:       oid_token_field
            )
          end

          it "does not raise an error" do
            expect { subject }.to_not raise_error
          end
        end

        context "with xms_mirid that does not begin with a /" do
          subject do
            Authentication::AuthnAzure::ValidateApplicationIdentity.new(
              resource_class:             resource_class,
              validate_azure_annotations: validate_azure_annotations,
              ).call(
              account:               account,
              service_id:            service_id,
              username:              username,
              xms_mirid_token_field: xms_mirid_token_missing_initial_slash,
              oid_token_field:       oid_token_field
            )
          end
          it "does not raise an error" do
            expect { subject }.to_not raise_error
          end
        end
      end
...
```

Reply:
Am reluctant do because it is a permutation of and xms mirid and should xms_mirid format